quix-frontend/client/src/lib/sql-autocomplete/adapter/sql-autocomplete-adapter-utills.ts

```diff
@@ -0,0 +1,164 @@
+
+  for (const table of tables) {
+    for (const column of table.columns) {
+      let found = false;
+      let gotRelevantPartOfPrefix = false;
+      let currentPrefix = '';
+
+      brokenPrefix.forEach(cell => {
+        if (found) {
+          currentPrefix += cell + '.';
+        }
+        if (typeof column === 'object' && !gotRelevantPartOfPrefix && cell === column.name) {
+          found = true;
+          gotRelevantPartOfPrefix = true;
+          currentPrefix += cell + '.';
+        }
+      });
+
+      if (currentPrefix) {
+        relevantPrefixes.push(currentPrefix);
+      }
+    }
+  }
+
+  if (relevantPrefixes.length === 0) {
+    return ''; // No relevant prefixes found
+  }
+
+  // Find the longest relevant prefix
+  let longestPrefix = relevantPrefixes[0];
+  for (const prefix of relevantPrefixes) {
+    if (prefix.length > longestPrefix.length) {
+      longestPrefix = prefix;
+    }
+  }
+
+  return longestPrefix
+}
+
+
+
+function getAllChildrenOfTables(tables: TableInfo[]) {
+  const allChildren = [];
+
+  for (const extractedTable of tables) {
+    const { columns } = extractedTable;
+
+    for (const column of columns) {
+      if (typeof column === 'object') {
+        if (typeof column.dataType === 'string') {
+          column.dataType = trinoToJs(column.dataType, 0);
+        }
+        if (typeof column.dataType === 'object') {
+          allChildren.push(...getObjectChildren(column.dataType, column.name));
+        }
+      }
+    }
+  }
+
+  return allChildren;
+}
+
+
+export function getSearchCompletion(tables: TableInfo[], prefix: string | undefined): any {
+  if (!prefix.includes('.')) {
+    return [];
+  }
+  const allChildren = getAllChildrenOfTables(tables);
+  const relevantPartOfPrefix = findRelevantPartOfPrefix(tables, prefix.split('.')).slice(0, -1);
+  if (!relevantPartOfPrefix) {
+    return []
+  }
+  const lastDotIndex = findLastDotIndex(relevantPartOfPrefix);
+  const startOfSearch = lastDotIndex !== -1 ? relevantPartOfPrefix.slice(0, lastDotIndex + 1) : relevantPartOfPrefix;
+  const searchPart = relevantPartOfPrefix.replace(startOfSearch, '')
+  const filteredChildren = allChildren.filter(obj => {
+    const lowerCasedName = obj.name.toLowerCase();
+    const parts = lowerCasedName.split('.');
+    if (parts.length > 1) {
+      const substringAfterFirstDot = parts.slice(1).join('.');
+      const criteria = checkCriteria(substringAfterFirstDot, searchPart.toLowerCase());
+      const filterIfInDifferentColumn = obj.name.startsWith(relevantPartOfPrefix.split('.')[0]);
+      return obj.name.includes(startOfSearch) && substringAfterFirstDot.includes(searchPart.toLowerCase()) && criteria && filterIfInDifferentColumn;
+    }
+    return false;
+  });
+  const prefixUntilLastDot = extractPrefixUntilLastDot(relevantPartOfPrefix);
+  const completionArray = filteredChildren.map(obj => ({
+    value: obj.name,
+    meta: typeof obj.dataType === 'object' ? 'row' : obj.dataType,
```

Comment:
aren't you supposed to fix it?

Reply:
anat pointed it out in the demo 
but we said we wanted to keep it consistent so we kept it row...
if you understood otherwise i can change to object

Reply:
my intention was - aren't we talked about that its supposed to be in a function?